In [ ]:
#1. Prepare your workstation¶

In [1]:
# Import the libarary
import pandas as pd

# Import the CSV files
oil = pd.read_csv('oil_price.csv')
gold = pd.read_csv('gold_stocks_price.csv')

# View the oil DataFrame
print(oil.shape)
print(oil.dtypes)
print(oil.head())
print(oil.tail())

(1844, 6)
Date      object
Price    float64
Open     float64
High     float64
Low      float64
Vol.      object
dtype: object
         Date   Price    Open    High     Low     Vol.
0  2011-12-01  100.20  100.51  101.17   98.87  275.86K
1  2011-12-02  100.96  100.00  101.56   99.76  245.92K
2  2011-12-05  100.99  101.23  102.44  100.24  239.38K
3  2011-12-06  101.28  100.45  101.42  100.20  217.03K
4  2011-12-07  100.49  101.11  101.94   99.67  330.53K
            Date  Price   Open   High    Low     Vol.
1839  2018-12-25  43.16  43.00  43.25  42.88        -
1840  2018-12-26  46.22  42.85  47.00  42.52  698.41K
1841  2018-12-27  44.61  46.58  46.70  44.37  665.59K
1842  2018-12-28  45.33  45.44  46.22  44.42  564.39K
1843  2018-12-31  45.41  45.22  46.53  44.73  476.97K


In [2]:
# View the gold DataFrame
print(gold.shape)
print(gold.dtypes)
print(gold.head())
print(gold.tail())

(1718, 81)
Date              object
Open             float64
High             float64
Low              float64
Close            float64
                  ...   
USO_High         float64
USO_Low          float64
USO_Close        float64
USO_Adj Close    float64
USO_Volume         int64
Length: 81, dtype: object
         Date        Open        High         Low       Close   Adj Close  \
0  2011-12-15  154.740005  154.949997  151.710007  152.330002  152.330002   
1  2011-12-16  154.309998  155.369995  153.899994  155.229996  155.229996   
2  2011-12-19  155.479996  155.860001  154.360001  154.869995  154.869995   
3  2011-12-20  156.820007  157.429993  156.580002  156.979996  156.979996   
4  2011-12-21  156.979996  157.529999  156.130005  157.160004  157.160004   

     Volume     SP_open     SP_high      SP_low  ...    GDX_Low  GDX_Close  \
0  21521900  123.029999  123.199997  121.989998  ...  51.570000  51.680000   
1  18124300  122.230003  122.949997  121.300003  ...  52.040001  52.6

In [3]:
oil["Date"] = pd.to_datetime(oil["Date"], errors="coerce")
gold["Date"] = pd.to_datetime(gold["Date"], errors="coerce")


In [4]:
# Gold subset (Date, Open, High, Low)

In [5]:
gold_subset = gold[["Date", "Open", "High", "Low"]].iloc[:500].copy()

print("gold_subset shape:", gold_subset.shape)
print(gold_subset.dtypes)
display(gold_subset.head())
display(gold_subset.tail())


gold_subset shape: (500, 4)
Date    datetime64[ns]
Open           float64
High           float64
Low            float64
dtype: object


Date        Open        High         Low
0 2011-12-15  154.740005  154.949997  151.710007
1 2011-12-16  154.309998  155.369995  153.899994
2 2011-12-19  155.479996  155.860001  154.360001
3 2011-12-20  156.820007  157.429993  156.580002
4 2011-12-21  156.979996  157.529999  156.130005

Date        Open        High         Low
495 2014-01-27  121.650002  122.059998  120.720001
496 2014-01-28  121.320000  121.400002  120.410004
497 2014-01-29  122.190002  122.510002  121.129997
498 2014-01-30  119.669998  120.220001  119.599998
499 2014-02-03  120.470001  122.110001  120.320000

In [ ]:
# Missing Values

In [6]:
print("Missing values (gold_subset):")
print(gold_subset.isna().sum())


Missing values (gold_subset):
Date    0
Open    0
High    0
Low     0
dtype: int64


In [ ]:
# Min/Max for Open/High/Low

In [7]:
print("Min/Max (gold_subset):")
display(gold_subset[["Open", "High", "Low"]].agg(["min", "max"]))


Min/Max (gold_subset):


Open        High         Low
min  117.580002  118.180000  116.739998
max  173.199997  174.070007  172.919998

In [ ]:
#  3) Oil subset (Date, Open, High, Low)

In [8]:
oil_subset = oil[["Date", "Open", "High", "Low"]].iloc[:500].copy()

print("oil_subset shape:", oil_subset.shape)
print(oil_subset.dtypes)
display(oil_subset.head())
display(oil_subset.tail())


oil_subset shape: (500, 4)
Date    datetime64[ns]
Open           float64
High           float64
Low            float64
dtype: object


Date    Open    High     Low
0 2011-12-01  100.51  101.17   98.87
1 2011-12-02  100.00  101.56   99.76
2 2011-12-05  101.23  102.44  100.24
3 2011-12-06  100.45  101.42  100.20
4 2011-12-07  101.11  101.94   99.67

Date   Open   High    Low
495 2013-11-13  93.07  94.54  92.93
496 2013-11-14  93.50  94.43  92.51
497 2013-11-15  93.89  94.55  93.58
498 2013-11-18  93.78  94.30  92.72
499 2013-11-19  93.00  93.46  92.43

In [ ]:
# Missing Values

In [9]:
print("Missing values (oil_subset):")
print(oil_subset.isna().sum())


Missing values (oil_subset):
Date    0
Open    0
High    0
Low     0
dtype: int64


In [ ]:
# Min/Max for Open/High/Low

In [10]:
print("Min/Max (oil_subset):")
display(oil_subset[["Open", "High", "Low"]].agg(["min", "max"]))


Min/Max (oil_subset):


Open    High     Low
min   78.11   79.68   77.28
max  110.28  112.24  109.11

In [ ]:
# 4) Statistical analysis (mean + 75th percentile via describe())

In [11]:
gold_stats = gold_subset[["Open", "High", "Low"]].describe(percentiles=[0.75])
oil_stats  = oil_subset[["Open", "High", "Low"]].describe(percentiles=[0.75])

display(gold_stats)
display(oil_stats)

# If you ONLY want mean + 75th percentile:
display(gold_stats.loc[["mean", "75%"]])
display(oil_stats.loc[["mean", "75%"]])


Open        High         Low
count  500.000000  500.000000  500.000000
mean   149.033480  149.774340  148.266740
std     16.923214   16.919180   16.871447
min    117.580002  118.180000  116.739998
50%    155.070000  155.610001  154.360001
75%    161.862500  162.577500  161.149994
max    173.199997  174.070007  172.919998

Open        High         Low
count  500.000000  500.000000  500.000000
mean    96.196620   97.153780   95.084320
std      6.991871    6.945661    7.003769
min     78.110000   79.680000   77.280000
50%     95.790000   96.640000   94.820000
75%    102.002500  102.962500  100.847500
max    110.280000  112.240000  109.110000

Open       High         Low
mean  149.03348  149.77434  148.266740
75%   161.86250  162.57750  161.149994

Open       High        Low
mean   96.19662   97.15378   95.08432
75%   102.00250  102.96250  100.84750


Missing values: both gold_subset and oil_subset have 0 missing values in Date/Open/High/Low (in the slices shown).

Gold (Open/High/Low): higher overall levels (mean Open ≈ 149; 75th percentile Open ≈ 161.86). The min/max range suggests noticeable movement across the period (Open min ≈ 117.58, max ≈ 173.20).

Oil (Open/High/Low): lower levels (mean Open ≈ 96.20; 75th percentile Open ≈ 102.00). Range is smaller in absolute terms (Open min ≈ 78.11, max ≈ 110.28).

Interpretation: in this window, gold trades at a higher price level and shows a wider absolute range; oil is lower-priced with a tighter absolute range. Next step would be to plot both over time to compare volatility properly (percent changes / returns are best for that).